### PR TITLE
update node20/esm

### LINKED ISSUE #63: Update actions to use `node20`

### LINKED ISSUE #74: node20 / esm / tag-release

### PR DESCRIPTION
**Pull request by le-cong**
Fixes #74 
Fixes #63

### CODE REVIEW

File: .github/workflows/check-published-scheduled.yml
Comment (carlansley) [inline .github/workflows/check-published-scheduled.yml:1]: I don't think we need this action.  its not a published npm module

File: .github/workflows/publish-beta.yml
Comment (carlansley) [inline .github/workflows/publish-beta.yml:1]: I don't think we need this action.  its not a published npm module

File: .github/workflows/publish.yml
Comment (carlansley) [inline .github/workflows/publish.yml:1]: I don't think we need this action.  its not a published npm module

File: package.json [hunk 1]
```diff
@@ -1,62 +1,74 @@
...
+    "nock": "^14.0.0-beta.5",
+    "rimraf": "^5.0.5",
+    "uuid": "^9.0.1"
   },
   "scripts": {
-    "dist": "rimraf dist && tsc --outDir dist && rimraf dist/*.spec.* dist/nocks",
-    "prepublishOnly": "npm run dist",
-    "lint": "eslint -f unix --ext .js,.ts src/**",
-    "lint:fix": "eslint -f unix --ext .js,.ts src/** --fix",
+    "prepublishOnly": "npm run build:dist-types && npm run build:dist-mjs",
```
Comment (carlansley) [inline package.json:61]: since this isn't being published, probably don't need types?

**Comment by le-cong** [comment]
@adcreare the Coverage job is failing and it seems to be caused by flaky tests and some other issue. is that sth we normally ignore? or there is sth we have to fix? - i did fixed one of the flaky test in check-label in case the PR label is not PATCH (MAJOR in this case)

**Review by carlansley — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (le-cong) [reply package.json:61]: in that case, i'll also remove the check-published check.

**Review by adcreare — APPROVED** [review]
LGTM

**Comment by adcreare** [comment]
I think the coverage issue is ok. I think its failing because it needs to move to node 20.

**Review by carlansley — APPROVED** [review]

**Pull request merged**